### DISCUSSION TITLE
Unable to Establish Serial communication

### DISCUSSION THREAD

**Discussion by UCmay** [sn-lab/MouseGoggles#15] [Q&A]
Sorry to disturb you, I would like to ask a question.
I follow your MouseVR protocol to build a mouse vr. I use ADSN3080 and Arduino DUE to detect running wheel movement. The code and wire connection are in the [image.](**url**) However, my computer cannot receive any data from DUE.
![ac0b83661e9bed2f0dc189a590f4780](https://github.com/sn-lab/mouseVRheadset/assets/140501075/0b316469-776f-4367-b621-d1f5f65bd86f)
![7777777](https://github.com/sn-lab/mouseVRheadset/assets/140501075/95fd4858-4a92-410e-a011-4adc75b68f2e)
Thank you.

**Comment by misaacson01** [comment]
I'm not sure what code that is, but it looks like the ADNS board isn't connected to the Due's SPI port. The SPI pins can't be assigned to just any pin, they're in the 6-pin connector near the center of the board ([right is SPI0 I believe](https://www.openinverter.org/wiki/images/thumb/a/ab/Due_SPI_pins.png/450px-Due_SPI_pins.png)). The RESET and CS pins can usually be assigned elsewhere, but MISO, MOSI, and SCK can't.

**Reply by UCmay** [reply]
1. We reconnect ADSN3080 to DUE. The connection map is shown below.
![aa0fc4727d23a33bd5e3001d643c723](https://github.com/sn-lab/mouseVRheadset/assets/140501075/6d9fc788-9673-47df-8581-3f53893f9594)
2. We upload ADSN3080 mousecamimages.ino from debugging program to our DUE. 
3. We run mouse_images.py. However, there is no image coming back and there is no data for mouse_motion.py either.
![8c96ba45f55bcdc3a408c579db570c2](https://github.com/sn-lab/mouseVRheadset/assets/140501075/f43db1a2-54a4-4a50-b665-4b8b7cfaa772)
![896b898fe88b355f022858f444400fa](https://github.com/sn-lab/mouseVRheadset/assets/140501075/27d713e8-edf0-43ba-80d5-0aa9fae03f8f)
Thanks.

**Reply by misaacson01** [reply]
Debugging with Python may be getting ahead of yourself since the Due wiring and program hasn't been validated yet, I would recommend debugging with just the Due and serial monitor. But, this might still be simple wiring mistakes. 5V is no longer connected to the 5V Arduino pin, and it looks like CS and RESET are on pins 4 and 5 when the default code has them on 10 and 11.

**Reply by UCmay** [reply]
Thank you for your reply. We have accept your suggestions. However, we still meet some problems. 
1. We connect wires again. 
![a668213def47a7cf5e8c1d7a1da5a6a](https://github.com/sn-lab/mouseVRheadset/assets/140501075/77ccecfb-66b6-48db-89bc-fb32a0dbb833)
2.We use the Due Programming Port to flash MouseCamImages.ino onto the Due.
![35add9aaa52443a65150a60744501bc](https://github.com/sn-lab/mouseVRheadset/assets/140501075/ea44b54c-0202-4600-862f-d0f8e378640f)
But it come back with Mouse cam failed to init. Is there a problem with our board, or is there an error in the wiring?
Thank you.

**Reply by misaacson01** [reply]
Can you try connecting the 3V pin as well? Either with or without the 5V pin connected. I haven't set up this board but it seems possible that the 3V is necessary to power the chip? Other than that, it looks right to me.

**Comment by UCmay** [comment]
Yes, we have tried to use the 3V pin. But it doesn't work either.
![d4fbffb968c952532c13399a1f7ab4e](https://github.com/sn-lab/mouseVRheadset/assets/140501075/cb1d16b6-f0f0-41df-adb9-bedeb34d1512)
![0438f85d768c29760d53f0f62910c55](https://github.com/sn-lab/mouseVRheadset/assets/140501075/f831575b-89bb-47b5-a4c6-4c8ec0e55e12)

**Reply by misaacson01** [reply]
Does it still say that the camera has failed to initialize? Sometime even when it's working it's hard to get a good picture out of it, but that's a separate issue. If it's still failed to initialize then I think it's reasonable to try another board.

**Comment by UCmay** [comment]
Maybe our board has some problems.
